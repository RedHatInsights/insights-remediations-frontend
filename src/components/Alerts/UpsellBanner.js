import React from 'react';
import propTypes from 'prop-types';

import { Alert, AlertActionCloseButton, Stack, StackItem } from '@patternfly/react-core';

const UpsellBanner = ({ onClose, ...props }) => {
    return (
        <Alert
            variant="info"
            isInline
            title="Do more with Find it Fix it capabilities"
            action={ <AlertActionCloseButton onClose={ onClose }/> }
            { ...props }>
            <Stack gutter='md'>
                <StackItem>Upgrade to Red Hat Smart Management to remediate all your systems, across regions and geographies</StackItem>
                <StackItem>
                    <a href="https://access.redhat.com/products/cloud_management_services_for_rhel/evaluation">Learn more</a>
                </StackItem>
            </Stack>
        </Alert>
    );
};

export default UpsellBanner;

UpsellBanner.propTypes = {
    onClose: propTypes.func
};

/* eslint-disable no-console */
UpsellBanner.defaultProps = {
    onClose: () => undefined
};
