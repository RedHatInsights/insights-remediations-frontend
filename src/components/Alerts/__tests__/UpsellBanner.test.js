import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import UpsellBanner from '../UpsellBanner';

describe('UpsellBanner component', () => {

    it('should render', () => {
        const wrapper = shallow(<UpsellBanner/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should call default prop', () => {
        const mockCallBack = jest.fn();
        const wrapper = mount(<UpsellBanner/>);
        wrapper.find('Button').simulate('click');
        expect(mockCallBack.mock.results[0]).toBeUndefined();
    });

    it('should click close', () => {
        const mockCallBack = jest.fn();
        const wrapper = mount(<UpsellBanner onClose={ mockCallBack }/>);
        wrapper.find('Button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });
});
