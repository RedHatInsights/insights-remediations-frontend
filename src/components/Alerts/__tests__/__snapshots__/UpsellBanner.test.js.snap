// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpsellBanner component should render 1`] = `
<Component
  action={
    <AlertActionCloseButton
      onClose={[Function]}
    />
  }
  isInline={true}
  title="Do more with Find it Fix it capabilities"
  variant="info"
>
  <Stack
    gutter="md"
  >
    <StackItem>
      Upgrade to Red Hat Smart Management to remediate all your systems, across regions and geographies
    </StackItem>
    <StackItem>
      <a
        href="https://access.redhat.com/products/cloud_management_services_for_rhel/evaluation"
      >
        Learn more
      </a>
    </StackItem>
  </Stack>
</Component>
`;
